### PR TITLE
Doc: LLNL Setups

### LINKED ISSUE #1132: Compiling WarpX for GPU on Lassen
I am trying to compile WarpX on Lassen, whose specifics can be found here: https://hpc.llnl.gov/hardware/platforms/lassen

The outputted error can be found in the attachment.

I was told that Lassen architecture would be closest to the instructions for Summit, but a lot of the modules listed are not present on our system. 

I noticed that you recommended gcc/6.4.0 but the only options on Lassen are 4.9.3, 7.2.1-redhat, 7.3.1, and 8.3.1, none of which seem to give successful compilations. On a different machine I could compile with Intel, but that is not an option on Lassen. The error report mentions a missing -lmpi_ibm_mpifh, which I am not sure where to find. For MPI there are mvapich2 and spectrum modules, but I am not sure what the default is on Summit.

Please let me know if you need more information, I am new to compiling on Lassen/GPU and am not sure myself what to look for.
[warpx_lassen_comp.txt](https://github.com/ECP-WarpX/WarpX/files/4848753/warpx_lassen_comp.txt)

### PR DESCRIPTION
**Pull request by ax3l**
- Lassen (LLNL): V100 (4x/node)
  - [x] Document installation and usage
  - [x] compile test
  - [x] runtime test
  - [ ] openpmd runtime test: waiting for HDF5 and/or ADIOS2 modules
  - [ ] performance tuning: hints for 2D3V and quasi 1D3D simulations (sorting, `amr.max_grid_size`, `amr.blocking_factor`, dynamic load balancing); V100/SM7.0 only (works, Coral login nodes have GPUs)
  - close #1132
- Quartz (LLNL): Intel Intel Xeon E5-2695 v4 CPU (1x/node)
  - [x] Document installation and usage
  - [x] compile test
  - [x] runtime test
  - [x] use newer libstdc++ that is not affected by #1396
  - [x] openpmd runtime test (HDF5): fix ICC bug(s) [[1]](https://github.com/openPMD/openPMD-api/pull/788) [[2]](https://github.com/openPMD/openPMD-api/pull/789)
  - [ ] performance tuning: double-check number of CPUs and if HT is really fast; hints for 2D3V and quasi 1D3D simulations (sorting, `amr.max_grid_size`, `amr.blocking_factor`, dynamic load balancing); `mtune`/`march`/`mcpu`; particles.tile_size

cc @bzdjordje

### CODE REVIEW

File: Docs/source/building/quartz.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+We use the following modules and environments on the system.
+
+.. code-block:: bash
+
+   # please set your project account
+   export proj=<yourProject>
+
+   # required dependencies
+   module load cmake/3.16.8
+   module load intel/19.0.4
```
Comment (ax3l) [inline Docs/source/building/quartz.rst]:
```suggestion
   module load intel/19.1.2
```

File: Docs/source/building/quartz.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+   module load mvapich2/2.3
+
+   # optional: for PSATD support
+   module load fftw/3.3.8
+
+   # optional: for QED support
+   module load boost/1.72.0
```
Comment (ax3l) [inline Docs/source/building/quartz.rst]:
```suggestion
   module load boost/1.73.0
```

File: Tools/BatchScripts/batch_quartz.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# the settings below make sure threads are close to the
+#   controlling MPI rank (process) per half socket and
+#   distribute equally over close-by physical cores and,
+#   for N>9, also equally over close-by logical cores
+export OMP_PROC_BIND=spread
+export OMP_PLACES=threads
+export OMP_NUM_THREADS=18
+
+# for async_io support: (optional)
+# TODO if needed
```
Comment (ax3l) [inline Tools/BatchScripts/batch_quartz.sh]: Works:
```
MPI initialized with thread support level 3
```
```suggestion
```

**Comment by ax3l** [comment]
Runtime tests successful. For more features, some module requests are still ongoing.
Open points will be addressed in follow-up PRs :)

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**